test/views/UnderpaymentSummaryViewSpec.scala

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views
+
+import base.ViewBaseSpec
+import messages.UnderpaymentSummaryMessages
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import play.api.mvc.Call
+import play.twirl.api.Html
+import views.data.UnderpaymentSummaryData._
+import views.html.UnderpaymentSummaryView
+
+class UnderpaymentSummaryViewSpec extends ViewBaseSpec {
+
+  private lazy val injectedView: UnderpaymentSummaryView = app.injector.instanceOf[UnderpaymentSummaryView]
+
+  private val backLink: Call = Call("GET", "url")
+
+  "Rendering the UnderpaymentSummary page" when {
+    "only 1 underpayment made" should {
+
+      lazy val view: Html = injectedView(None, importVat, None, backLink)(fakeRequest, messages)
+      lazy implicit val document: Document = Jsoup.parse(view.body)
+
+      "have only 1 sub-heading" in {
+        document.select("main h2").size mustBe 1
```

Comment:
"main h2" and the other selectors are duplicated quite a bit

Reply:
I agree there is duplication in the test for extracting these elements from the document but I find it easier to understand where the data has come from if I can see the selector in the test.